bindings/dart/lib/src/common/reranker/ai.dart

```diff
@@ -76,15 +77,14 @@ class XaynAi {
   void free() => throw UnsupportedError('Unsupported platform.');
 }
 
+/// Maximum number of threads to be used for parallel features.
+const int maxNumberOfThreads = 16;
+
 /// Selects the number of threads used by the [`XaynAi`] thread pool.
 ///
 /// On a single core system the thread pool consists of only one thread.
 /// On a multicore system the thread pool consists of
-/// (the number of logical cores - 1) threads.
-int selectThreadPoolSize(int numberOfProcessors) {
-  if (numberOfProcessors > 1) {
-    return numberOfProcessors - 1;
-  } else {
-    return numberOfProcessors;
-  }
-}
+/// (the number of logical cores - 1) threads, but at most [`maxNumberOfThreads`] threads.
+int selectThreadPoolSize(int numberOfProcessors) => (numberOfProcessors > 1)
```

Comment:
Don't know if it will be still good for readability, but we can simplify it further as:
`min(max(numberOfProcessors - 1, 1), maxNumberOfThreads)`.